server_http.hpp

```diff
@@ -217,7 +217,8 @@ namespace SimpleWeb {
         
         ///Use this function if you need to recursively send parts of a longer message
         void send(const std::shared_ptr<Response> &response, const std::function<void(const boost::system::error_code&)>& callback=nullptr) const {
-            boost::asio::async_write(*response->socket, response->streambuf, [this, response, callback](const boost::system::error_code& ec, size_t /*bytes_transferred*/) {
+            boost::asio::async_write(*response->socket, response->streambuf, [response, callback](const boost::system::error_code& ec, size_t /*bytes_transferred*/) {
+                (void) response; // response is not used here, but needs to captured to keep it alive because async_write is using response->*-fields
```

Comment:
That is a good argument actually, but not sure if `-Wunused-lambda-capture` should be activated with `-Wall` or `-Wextra`. It might be an oversight by the clang++ developers, since using captures to prolong an shared_ptr's lifetime should be a valid use case, especially together with Boost.Asio. Not sure if the asio proposal to the C++ standard solves this in a different way. I'll have a look at it Tomorrow.